### DISCUSSION TITLE
Where text array contains

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#8478] [Questions] [answered]
Text Array Column value:

```
[
  [
    "banana",
    "orange"
  ],
  [
    "red",
    "blue"
  ]
]
```

Trying to figure out the correct formatting for query to find any rows where searchstring is included anywhere in the Text Array Column. eg %anan% returns above row.

Tried for example:

`.contains('Text Array Column','%'+searchstring+'%')
`

**Comment by GaryAustin1** [comment, marked as answer]
You might be able to use a computed column (postgres function to generate a virtual column) https://postgrest.org/en/stable/api.html#computed-columns that is a concatenated string and then do ilike/like filter on that.  

Otherwise you would need to do an rpc call to a Postgres function to do an array searchon partial strings that is a bit complicated.  If you google something like "postgres array string search partial".  An example: https://stackoverflow.com/questions/30096992/is-there-a-way-to-search-partial-match-in-multivalue-field-in-postgresql
